File: src/entities.rs
use chrono::Utc;
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Serialize, Deserialize)]
pub struct Cohort {
    pub id: Uuid,
    pub name: String,
    pub email: String,
    pub created_at: chrono::DateTime<Utc>,
    pub updated_at: chrono::DateTime<Utc>,
}

#[derive(Serialize, Deserialize)]
pub struct Student {
    pub id: Uuid,
    pub name: String,
    pub number: String,
    pub created_at: chrono::DateTime<Utc>,
    pub updated_at: chrono::DateTime<Utc>,
}

#[derive(Serialize, Deserialize)]
pub struct Assignment {
    pub id: Uuid,
    pub name: String,
    pub description: String,
    pub cohort_id: Uuid,
    pub created_at: chrono::DateTime<Utc>,
    pub updated_at: chrono::DateTime<Utc>,
}

#[derive(Serialize, Deserialize)]
pub struct AssignmentScore {
    pub assignment_id: Uuid,
    pub student_id: Uuid,
    pub score: i32,
    pub created_at: chrono::DateTime<Utc>,
    pub updated_at: chrono::DateTime<Utc>,
}

File: web/src/lib/entities.ts
export interface Cohort {
    id: string;
    name: string;
    email: string;
    created_at: string;
    updated_at: string;
}

export interface Student {
    id: string;
    number: string;
    name: string;
    cohort_id: string;
    created_at: string;
    updated_at: string;
}

export interface Assignment {
    id: string;
    name: string;
    description: string;
    cohort_email: string;
    cohort_name: string;
    cohort_id: string;
    created_at: string;
    updated_at: string;
}

export interface AssignmentScore {
    assignment_id: string;
    assignment_name: string;
    cohort_id: string;
    cohort_name: string;
    student_id: string;
    student_number: string;
    student_name: string;
    score?: number;
    created_at?: string;
    updated_at?: string;
}

export interface Wrapper<T> {
    data: T;
    error: any;
    message: string;
}
